Mage/src/main/java/mage/game/GameImpl.java

```diff
@@ -2498,6 +2499,31 @@ protected boolean checkStateBasedActions() {
                     somethingHappened = true;
                 }
             }
+
+            if (perm.isBattle(this)) {
+                if (perm
+                        .getCounters(this)
+                        .getCount(CounterType.DEFENSE) == 0
+                        && this.getStack()
+                        .stream()
+                        .filter(StackAbility.class::isInstance)
+                        .filter(stackObject -> stackObject.getStackAbility() instanceof TriggeredAbilityImpl)
+                        .map(StackObject::getSourceId)
+                        .noneMatch(perm.getId()::equals)
+                        && this.state
+                        .getTriggered(perm.getControllerId())
+                        .stream()
+                        .filter(TriggeredAbility.class::isInstance)
+                        .map(Ability::getSourceId)
+                        .noneMatch(perm.getId()::equals)) {
+                    movePermanentToGraveyardWithInfo(perm);
+                    somethingHappened = true;
```

Comment:
Must be
``` java
if (movePermanentToGraveyardWithInfo(perm)) {
    somethingHappened = true;
}
```

Reply:
I don't think it makes a difference but I'll change it